CppCoreGuidelines.md

```diff
@@ -20029,9 +20029,95 @@ Following this rule leads to weaker invariants,
 more complicated code (having to deal with semi-constructed objects),
 and errors (when we didn't deal correctly with semi-constructed objects consistently).
 
-##### Example
+##### Example, bad
 
-    ???
+    class Picture
+    {
+        int mx;
+        int my;
+        char * data;
+    public:
+        Picture(int x, int y)
+        {
+            mx = x,
+            my = y;
+            data = nullptr;
+        }
+    
+        ~Picture()
+        {
+            Cleanup();
+        }
+    
+        bool Init()
+        {
+            // invariant checks
+            if (mx <= 0 || my <= 0 || mx % 2 != 0 || my % 2 != 0) {
+                return false;
+            }
+            if (data) {
+                return false;
+            }
+            data = (char*) malloc(x*y*sizeof(int));
+            return data != nullptr;
+        }
+    
+        void Cleanup()
+        {
+            if (data) free(data);
+            data = nullptr;
+        }
+    };
+    
+    Picture picture(3, 5); // not ready-to-use picture here
+    // this will fail..
+    if (!picture.Init()) {
+        puts("Error, invlaid picture");
+    }
+    // now have a invalid picture object instance.
+
+##### Example, good
+
+    class Picture
+    {
+        int mx;
+        int my;
```

Comment:
don't forget defaults (C.43 and C.48)

Reply:
As I thought about to change it, I realized a problem: If the default is 0, it will be a value not passing the invariant check. But if the default value is greater then 0, I should better also allocate some memory for the data vector. So, I think this class must be initialized with the "size" constructor always. Any thoughts?

Reply:
Is it still a problem? If so, what is the suggested change? I think, the 3 members can only be set via constructor, because they are depended.